### PR TITLE
Add environment UUID support to socket requests

### PR DESCRIPTION
**Pull request by hatched**
The newer versions of Juju core support us sending the environment UUID in the socket requests. This branch also removes some old unsupported configuration options from the socket attributes.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2370/
Test PASSed.

File: app/app.js [hunk 1]
```diff
@@ -701,6 +684,36 @@ YUI.add('juju-gui', function(Y) {
...
+      if (loc.port) {
+        socketUrl += ':' + loc.port;
+      }
+      // If the Juju version is over 1.21 then we need to make requests to the
+      // api using the environments uuid.
+      var jujuVersion = this.get('jujuCoreVersion').split('.');
+      var suffix = '';
+      var majorVersion = parseInt(jujuVersion[0], 10);
+      var minorVersion = parseInt(jujuVersion[1], 10);
+      if (majorVersion === 1 && minorVersion > 20 || majorVersion > 1) {
```
Comment (makyo) [inline app/app.js]: I suppose this is more robust than simply checking if jujuEnvUUID is set?

Comment (hatched) [inline app/app.js]: Yeah the UUID has actually been available for some time, much longer than the api supported it unfortunately.

**Comment by makyo** [comment]
:+1: Spun up locally, deployed some services, and watched the websocket connection, everything worked fine

**Comment by huwshimi** [comment]
Code looks good, no QA. :+1: Thanks @hatched!

**Comment by hatched** [comment]
Thanks for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**